src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/Table.js

```diff
@@ -1,22 +1,136 @@
 // @flow
+import {action, observable} from 'mobx';
+import {observer} from 'mobx-react';
 import React from 'react';
-import type {TableChildren} from './types';
+import type {Element} from 'react';
+import Icon from '../Icon';
+import Header from './Header';
+import Body from './Body';
+import type {ControlItems, SelectMode, TableChildren} from './types';
 import tableStyles from './table.scss';
 
+const PLACEHOLDER_ICON = 'battery-quarter';
+
 type Props = {
     children: TableChildren,
+    /** List of buttons to apply action handlers to every row (e.g. edit row) */
+    controls?: ControlItems,
+    /** Can be set to "single" or "multiple". Defaults is "none". */
+    selectMode?: SelectMode,
+    /** 
+     * Callback function to notify about selection and deselection of a row.
+     * If the "id" prop is set on the row, the "rowId" corresponds to that, else it is the index of the row.
+     */
+    onRowSelectionChange?: (rowId: string | number, selected?: boolean) => void,
+    /** Called when the "select all" checkbox in the header was clicked. Returns the checked state. */
+    onAllSelectionChange?: (checked: boolean) => void,
+    /** Text shown when the table has no entries */
+    placeholderText?: string,
 };
 
+@observer
 export default class Table extends React.PureComponent<Props> {
+    static defaultProps = {
+        selectMode: 'none',
+    };
+
+    @observable allSelected = false;
+
+    @action setAllSelected = (allSelected: boolean) => {
```

Comment:
I think it is weird that this is a action... Why not just checking all rows and infer it from there? This way it might just end up having the wrong value...